html/settings.php

```diff
@@ -1,11 +1,17 @@
 <?php
 
+require_once "../php/config.php";
 require_once "../php/locale.php";
 require_once "../php/db_pdo.php";
 require_once "../php/helper.php";
 
 $type = isset($_GET["new"]) ? "signup" : "settings";
 
+if (isset($OF_ENABLE_SIGNUP) && !$OF_ENABLE_SIGNUP) {
```

Comment:
This reminds me of register_globals.

Definitely feels like we should be loading in a "default" config before loading config.php ontop.

Doesn't need fixing here though.

Reply:
Yeah, that would be nice. I went with the `isset` approach for backwards-compatibility but I think we've already broken it with #1267.

A configuration module/class would allow defining more than just variables/constants. That would be the only thing reading `config.php` and we could deal with compatibility there.